### ISSUE TITLE
Ensure that we render a confirmation modal on the toggle change for the Spaces page

### ISSUE THREAD

**Issue by jeremyf** [forem/forem#17032]
The goal of the confirmation modal is to highlight the "destructive" nature of the change.  A simple "Are you sure?" will not be adequate.  We should restate the impact of the change (with language based on the conditional).

As previously written, we are to remove the submit button, but in conversations with @aitchiss we should keep the submit button for accessibility purposes.

**Comment by amywtlin** [comment]
@anujbhavsar96 Jeremy has created a new issue here to track [this](https://github.com/forem/forem/issues/16907#issuecomment-1080476135). Let's continue the discussion here.

Summarizing the approach confirmed so far:

1. We will use [the browser modal](https://forem-team.slack.com/archives/C7GE84DEJ/p1648493559735239) to double confirm the action
2. We'll add a flash message after the change has been saved.
4. We will put information of 1. running campaign will be hidden and 2. orgs will be disabled before admins hit `save` button, following Anuj's design pattern [here](https://github.com/forem/forem/issues/16907#issuecomment-1080476135):
![image](https://user-images.githubusercontent.com/11730007/160759608-be276dfb-174f-499b-94a6-3f3045904535.png) 

Proposed copies

1. Browser modal text:
   > Are you sure you want to turn on comment-only for this space?
1. information section:
   > ℹ️ **If you turn on comment-only for this space, it also means that for the time being...**
   > -  Your active campaign will be hidden. Learn more about [Campaigns](https://admin.forem.com/docs/advanced-customization/config/campaigns)
   > -  Organizations will also be disabled. Learn more about [Organizations](https://admin.forem.com/docs/managing-your-community/organization-pages)
1. Flash message text
    > Comment-only has been turned on for {space name} successfully.

@juliannatetreault Feel free to review the copies above if you have a chance, and let me know what you think! 
Re: flash messages, we didn't really have an established pattern yet so I am following a similar pattern of the ones we have on member detail view right now:

<img width="1380" alt="Screen Shot 2022-03-30 at 1 58 39 PM" src="https://user-images.githubusercontent.com/11730007/160761567-6d820d97-9963-4faa-ae0f-cd937892b096.png">



@jeremyf is there any other implications you are wrestling that Anuj and I can help with?

**Comment by aitchiss** [comment]
Just to follow up on the slack conversation the form in the design above will need to have a "Save"/"Submit" button too - clicking that can trigger the browser alert.

(I'm not sure if the original intention was to have the alert appear when the user clicks on the toggle button, but this would break some of the web content accessibility guidelines criteria)

**Comment by anujbhavsar96** [comment]
That alert browser modal supposes to work as Save/submit for this toggle action.

Having Save/submit on the form level and again in the alert sounds confusing. Since this is a critical action we decided to have a modal.

**Comment by aitchiss** [comment]
Triggering a 'window confirm' (i.e. browser default confirmation pop up) on a toggle button press breaks WCAG success criterion [3.2.2: On Input](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html) in that it's not an expected interaction pattern for clicking a toggle button or a form control generally.

We should also be [including a submit button for any form](https://www.w3.org/WAI/WCAG21/Techniques/html/H32), again to maintain predictable behaviour. I believe this is partly why we use the crayons-modal in most newer user flows, as it allows to add an explicit submit button as part of the confirmation.

**Comment by aitchiss** [comment]
I don't know if a compromise could be to change the toggle button to a submit button with text e.g. "Enable"/"Disable", which could then trigger the window confirm popup? Then we have a submit button, but the user still only needs to click once to trigger the confirm flow

**Comment by anujbhavsar96** [comment]
> I don't know if a compromise could be to change the toggle button to a submit button with text, e.g., "Enable"/"Disable."

That's also kind of 1 or 0 kinds of action. Similar to toggle🤔. 

And according to [this](https://www.w3.org/WAI/WCAG21/Techniques/general/G13.html), we have all the instructions on the place about what changes will occur. 

About the submit button, can't we consider the confirmation modal a submit form?

**Comment by anujbhavsar96** [comment]
> Triggering a 'window confirm' (i.e. browser default confirmation pop up) on a toggle button press breaks WCAG success criterion [3.2.2: On Input](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html) in that it's not an expected interaction pattern for clicking a toggle button or a form control generally.

Just to not break the interaction pattern, should we inform them upfront that, they will be prompt with a confirmation popup or something? Not sure about this, saying it by reading [this](https://www.w3.org/WAI/WCAG21/Techniques/failures/F37.html)

**Comment by aitchiss** [comment]
> Just to not break the interaction pattern, should we inform them upfront that, they will be prompt with a confirmation popup or something? Not sure about [this](https://www.w3.org/WAI/WCAG21/Techniques/failures/F37.html), saying it by reading this

Yeah if we add text that says they'll be prompted to confirm on selection then we can meet this criteria I think

> About the submit button, can't we consider the confirmation modal a submit form?

Not if we use the browser window confirm, because we don't have any control of what's rendered in there apart from the text content 😞 . If we use a crayons modal we can insert a submit button as the confirm button. If we use the crayons modal there's no obstacle to it, and it's a pattern we have in various other places.

**Comment by aitchiss** [comment]
> Ah, I see. That means the problem is the browser confirmation popup. If we use the crayon modal then it's fine. In that case should we build the crayon modal then?👀

I think it'd be very straightforward / minimal extra effort 👍

**Comment by aitchiss** [comment]
Just to summarise what I _think_ we've agreed (please correct me if wrong) -

- A change to copy to include something about the fact that clicking the toggle button will trigger a confirm 
- Implement the crayons-modal so that when I click the toggle button, I see the modal, and clicking 'Confirm' on that modal submits the form

If that's right, I just have one other question so that I can assist @jeremyf with the frontend of this - do we currently have this kind of toggle button style implemented elsewhere in the app (I mean the on/off 'switch' style)? 

I didn't see it in Storybook and I can't recall spotting it elsewhere (I might just not be remembering though) - just trying to capture if part of this work will also involve creating that new form component.

**Comment by amywtlin** [comment]
@aitchiss Any suggestion on that change of copy? @anujbhavsar96 and I have decided where to put that description, but whatever I wrote feels awkward and unnatural. I have not seen other applications using this type of language before - maybe you can enlighten us?

**Comment by anujbhavsar96** [comment]
> do we currently have this kind of toggle button style implemented elsewhere in the app (I mean the on/off 'switch' style)?

Also @aitchiss we don't really have a toggle implemented anywhere. Probably @ludwiczakpawel can answer this.

**Comment by aitchiss** [comment]
@amy yeah it's hard for this particular situation. What do you think about:

> Only admins can publish posts. Other members can only react to and comment on posts. You will be asked to confirm any changes to this setting in a pop-up.


> Also @aitchiss we don't really have a toggle implemented anywhere. Probably @ludwiczakpawel can answer this.

Cool, so this issue will also include creating this crayons form element 👍

**Comment by amywtlin** [comment]
> Only admins can publish posts. Other members can only react to and comment on posts. You will be asked to confirm any changes to this setting in a pop-up.

This looks good to me!

**Comment by anujbhavsar96** [comment]
Made these copy [changes in Figma](https://www.figma.com/file/SeTWMa3E7F7fieVKm2wvCt/Roles-and-Authorization-System?node-id=309%3A3716)

**Comment by jeremyf** [comment]
My proposal is this: I'm going to push up the accessible no modal version with the confirmation dialogue; it will not include the URLs nor the bullet list (as I don't know what those URLs would be) nor will it include the new footer.

From there, we can go down the path of writing up issues to address these things:

1) Adjusting copy
2) Tracking down the URLs and adding to copy.
3) Determining the modal behavior because of a new understanding I have.

I also want to make sure that @Ridhwana can weigh in on the technical implications of adding modals (versus leveraging the existing flash system) and the effort as it all relates to our other goals.

**Comment by amywtlin** [comment]
@jeremyf Your proposed approach sounds fine to me. I don't think this is a blocker, but we should eventually get to that state to reflect Anuj's latest design & accessibility copy. 

The regression was introduced because we were looking for ways to address the issue you brought up where we shouldn't have new information in a popup, which makes sense to me. The final iteration is to show that information up front on the main page, so the confirm popup is simply to confirm the action, with no additional information.

**Comment by amywtlin** [comment]
@JennieOcken this is an issue where we are having an active discussion for Roles & Authorization System project. Please help keep an eye on this one when I am on leave, in case teammates can use your help to unstick/unblock while I am not around.

**Comment by jeremyf** [comment]
My plan is to write-up a new issue for the next step of this work.  Those next steps are as follows:

- [ ] Get the URLs that are implied in Anuj's [figma design](https://www.figma.com/file/SeTWMa3E7F7fieVKm2wvCt/Roles-and-Authorization-System?node-id=309%3A3716)
- [ ] Adjust copy to reflect Anuj's design but account for the regression
- [ ] Add modal behavior (this will be a @aitchiss) task but one that Jeremy would happily pair up on.

I want to hold off on converting that to an issue until after @Ridhwana has a chance to read through this thread.

**Comment by Ridhwana** [comment]
> Get the URLs that are implied in Anuj's [figma design](https://www.figma.com/file/SeTWMa3E7F7fieVKm2wvCt/Roles-and-Authorization-System?node-id=309%3A3716)

I suspect that Campaigns = https://admin.forem.com/docs/advanced-customization/config/campaigns, and Organizations = https://admin.forem.com/docs/managing-your-community/organization-pages#overview.

> I also want to make sure that @Ridhwana can weigh in on the technical implications of adding modals (versus leveraging the existing flash system) 
@jeremyf I read through the thread and just to clarify that I understand what you mean, are you talking about the technical implications of adding modals (versus _leveraging the default browser window confirm method_)? or did you actually mean flash system which is the little notices that appear at the top of the page?

In general this is a pattern that I see us using on the admin more often:
> Implement the crayons-modal so that when I click the toggle button, I see the modal, and clicking 'Confirm' on that modal submits the form.
and one that I will most likely addd to the other places where we need a double confirmation like Extensions etc.

**Comment by anujbhavsar96** [comment]
> Get the URLs that are implied in Anuj's 

The URLs Ridhwana provided are correct. I think we will have to add some information in the admin guide regarding this on both pages. Explaining changes if comment-only is turned on. @amywtlin, what do you think?

> but account for the regression in organization language (@anujbhavsar96 could do that in Canva in consultation with @amywtlin though Jeremy is prepared to provide guidance)

1. Only admins will be able to post on behalf of organizations. [Learn more](https://admin.forem.com/docs/managing-your-community/organization-pages#overview)
2. Only admins who are part of organization will be able to post on behalf of organization. [Learn more](https://admin.forem.com/docs/managing-your-community/organization-pages#overview)

We had decided on the first option but the first one is a bit unclear. It looks like the admin will be able to post on behalf of any organization even if they are not part of that organization. @jeremyf what do you think?

**Comment by jeremyf** [comment]
> > Get the URLs that are implied in Anuj's [figma design](https://www.figma.com/file/SeTWMa3E7F7fieVKm2wvCt/Roles-and-Authorization-System?node-id=309%3A3716)
> 
> I suspect that Campaigns = https://admin.forem.com/docs/advanced-customization/config/campaigns, and Organizations = https://admin.forem.com/docs/managing-your-community/organization-pages#overview.
> 
> > I also want to make sure that @Ridhwana can weigh in on the technical implications of adding modals (versus leveraging the existing flash system)
> 
> @jeremyf thanks! I read through the thread and just to clarify that I understand what you mean, are you talking about the technical implications of adding modals (versus _leveraging the default browser window confirmation)? or did you actually mean flash system which is the little notices that appear at the top of the page?
> 
> In general adding modals is definitely a pattern that I see us using on the admin more often:
> 
> > Implement the crayons-modal so that when I click the toggle button, I see the modal, and clicking 'Confirm' on that modal submits the form.
> 
> and one that I will most likely addd to the other places where we need a double confirmation like Extensions etc.

My main concern is how much of the UI requests are nice to have versus blockers.  As merged, the code works with minimal moving parts.  Adding a modal will add moving parts.  And I'm looking at my delivery date and wondering and assuming:

1. Updating copy is a blocker to "done" (e.g. we should provide context)
2. Adding a modal is a non-blocker to "done"

**Comment by anujbhavsar96** [comment]
Your assumptions are correct. I don't think the confirmation modal is a major blocker
